test/schema.tests.js

```diff
@@ -0,0 +1,135 @@
+var jwt = require('../index');
+var expect = require('chai').expect;
+var fs = require('fs');
+
+describe('schema', function() {
+
+  describe('sign options', function() {
+
+    var cert_priv = fs.readFileSync(__dirname + '/rsa-private.pem');
+
+    function sign(options) {
+      jwt.sign({foo: 123}, cert_priv, options);
+    }
+
+    it('should validate expiresIn', function () {
+      expect(function () {
+        sign({ expiresIn: '1 monkey' });
+      }).to.throw(/"expiresIn" should be a number of seconds or string representing a timespan/);
+      expect(function () {
+        sign({ expiresIn: 1.1 });
+      }).to.throw(/"expiresIn" should be a number of seconds or string representing a timespan/);
+      sign({ expiresIn: '10s' });
+      sign({ expiresIn: 10 });
+    });
+
+    it('should validate notBefore', function () {
+      expect(function () {
+        sign({ notBefore: '1 monkey' });
+      }).to.throw(/"notBefore" should be a number of seconds or string representing a timespan/);
+      expect(function () {
+        sign({ notBefore: 1.1 });
+      }).to.throw(/"notBefore" should be a number of seconds or string representing a timespan/);
+      sign({ notBefore: '10s' });
+      sign({ notBefore: 10 });
+    });
+
+    it('should validate audience', function () {
+      expect(function () {
+        sign({ audience: 10 });
+      }).to.throw(/"audience" must be a string or array/);
+      sign({ audience: 'urn:foo' });
+      sign({ audience: ['urn:foo'] });
+    });
+
+    it('should validate algorithm', function () {
+      expect(function () {
+        sign({ algorithm: 'foo' });
+      }).to.throw(/"algorithm" must be a valid string enum value/);
+      sign({algorithm: 'RS256'});
+      sign({algorithm: 'RS384'});
+      sign({algorithm: 'RS512'});
+      // not sure how to fix error - 'Could not find expected "seq"'
```

Comment:
Wasn't sure how to get these tests to pass, I assume I'm missing some required configuration for the `ES*` algorithms.